frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1387,8 +1387,17 @@ func dispatchCmd(d *dispatchState, c *instructions.CmdCommand) error {
 	if c.PrependShell {
 		args = withShell(d.image, args)
 	}
+	if d.image.OS == "windows" {
+		argsEscaped := c.PrependShell
+		// We warn here as Windows shell processing operates differently to Linux.
+		// Linux:   /bin/sh -c "echo hello" world   --> hello
+		// Windows: cmd /s /c "echo hello" world    --> hello world
+		if d.image.Config.ArgsEscaped != argsEscaped && len(d.image.Config.Entrypoint) > 0 {
+			// TODO error or warning about mixing CMD and ENTRYPOINT having unexpected results: https://github.com/moby/moby/blob/b8aa8579cad7b9c8712327093cbc602cfb6b417f/builder/dockerfile/dispatchers.go#L443
+		}
+		d.image.Config.ArgsEscaped = argsEscaped //nolint:staticcheck // ignore SA1019: field is deprecated in OCI Image spec, but used for backward-compatibility with Docker image spec.
+	}
 	d.image.Config.Cmd = args
-	d.image.Config.ArgsEscaped = true //nolint:staticcheck // ignore SA1019: field is deprecated in OCI Image spec, but used for backward-compatibility with Docker image spec.
```

Comment:
Yes, it absolutely does -- that was the original intent of my looking into this in the first place.  In short, `ArgsEscaped` should _never_ be set on a non-Windows image, and certainly never to `true`, and any case where it is gets (correctly) ignored by the runtimes anyhow.  It's a 100% Windows-specific field.

Reply:
The `ArgsEscaped` field was added in containerd [a while ago](https://github.com/containerd/containerd/pull/6479) and it essentially converts the behavior of the `Cmd` field into what `CmdLine` does (roughly). If `ArgsEscaped` is set to `true`, the shim expects that `ENTRYPOINT` or `CMD` is a single element array, with the command, along with any other arguments, already escaped.

For example, if we have `ArgsEscaped` set to `true`, by the time the `CMD` reaches the shim, it should look like this:

```Dockerfile
CMD ["\"c:\\Path to\\Some\\application.exe\" "some space delimited arg" someOtherArg"]
```

And that gets passed along to `HcsCreateProcess`. I remember having to deal with double escaping of args when I first implemented the windows executor. Back then we were sending `spec.Cmd` along with args. And the args were getting escaped again once they got to the shim. Details are fuzzy, but the consensus when I discussed this issue with the MSFT folks was to just send the one string as `CmdLine` instead of `Cmd` and `Args`.

I think this change is okay, as long as we just log the inconsistency in place of that TODO, similarly to what happens in moby now.